srcs/builtin/ft_exit.c

```diff
@@ -3,56 +3,45 @@
 #include "ft_builtin.h"
 #include "ft_dprintf.h"
 #include "minishell.h"
+#include "ft_string.h"
 
 // {"exit", "valid_arg", "invalid_arg1", "invalid_arg2", ..., NULL};
-// todo: naming...
-#define EXIT_ONLY_CMD_CNT		1
-#define EXIT_ARG_IDX			1
-#define VALID_ARG_CNT			2
 
-#define NON_NUMERIC_ARG_STATUS	2
-#define TOO_MANY_ARG_STATUS		1
-
-// todo: libft ...?
-static size_t	cnt_2d_array(const char **arr)
+static t_exit_arg	validate_argument(const char **cmds)
 {
-	size_t	i;
+	long	long_num;
+	size_t	argc;
+	bool	is_legal_num;
 
-	i = 0;
-	while (arr && arr[i])
-		i++;
-	return (i);
+	argc = count_2d_array(cmds);
+	if (argc == EXIT_ONLY_CMD_CNT)
+		return (EXIT_VALID_ARG);
+	is_legal_num = ft_legal_number(cmds[EXIT_ARG_IDX], &long_num);
+	if (!is_legal_num)
+		return (EXIT_NON_NUMERIC_ARG);
+	if (argc > VALID_ARG_CNT)
+		return (RETURN_TOO_MANY_NUMERIC_ARG);
+	return (EXIT_VALID_ARG);
 }
 
 // ft_legal_number returns true
-static int	get_exit_status(const char **cmds, \
-							t_exit_arg *res, int latest_status)
+static int	get_exit_status(const char *arg, \
+							t_exit_arg res, \
+							int latest_status)
 {
-	const size_t	argc = cnt_2d_array(cmds);
-	long			long_num;
-	bool			is_legal_num;
+	long	long_num;
 
-	if (argc == EXIT_ONLY_CMD_CNT)
-	{
-		*res = EXIT_VALID_ARG;
-		return (latest_status);
-	}
-	is_legal_num = ft_legal_number(cmds[EXIT_ARG_IDX], &long_num);
-	if (!is_legal_num)
-	{
-		*res = EXIT_NON_NUMERIC_ARG;
-		return (NON_NUMERIC_ARG_STATUS);
-	}
-	if (argc > VALID_ARG_CNT)
-	{
-		*res = RETURN_TOO_MANY_NUMERIC_ARG;
+	if (res == RETURN_TOO_MANY_NUMERIC_ARG)
 		return (TOO_MANY_ARG_STATUS);
-	}
-	*res = EXIT_VALID_ARG;
-	return ((int)(long_num & 255));
+	if (res == EXIT_NON_NUMERIC_ARG)
+		return (NON_NUMERIC_ARG_STATUS);
+	if (!arg)
+		return (latest_status);
+	ft_legal_number(arg, &long_num);
+	return ((int)(long_num & ((1U << BIT_SHIFT_FOR_STATUS) - 1)));
```

Comment:
こんな感じで説明変数に入れるとかはどうでしょう…？
```c
const unsigned int	bit_mask = (1U << BIT_SHIFT_FOR_STATUS) - 1;
```
```suggestion
	return ((int)(long_num & bit_mask));
```

Reply:
https://github.com/habvi/42_minishell/pull/77/commits/9580a18220ee78f1fe8e5a7e532b913f3854ebf5
# define BYTE_MASK  0xFF
を使ってみました！